collector/datacollector/datacollector.go

```diff
@@ -58,10 +67,51 @@ func main() {
 		glog.Exitf("Unable to obtain Log metadata: %s", err)
 	}
 
+	var ca *certgen.CA
+	if *addChainPeriod > 0 {
```

Comment:
Can you split this entire block out to a helper function that returns `(*certgen.CA, error)` to keep the `main` small and focussed?  Remove the `glog.Exitf()` from there, and just have one at the call-site in `main`.  Have the helper return `fmt.Errorf()` errors instead.

Reply:
Done, I think - check it is as you expected?

Reply:
LGTM